src/spec_swagger/swagger2/core.cljc

```diff
@@ -1,6 +1,94 @@
 (ns spec-swagger.swagger2.core
   (:require [clojure.walk :as walk]
-            [spec-swagger.swagger2.schema :as schema]))
+            [clojure.spec.alpha :as s]
+            [spec-tools.json-schema :as json-schema]
+            [spec-tools.visitor :as visitor]
+            [spec-tools.core :as st]))
+
+;;
+;; conversion
+;;
+
+(defn- spec-dispatch [dispatch _ _ _] dispatch)
+(defmulti accept-spec spec-dispatch :default ::default)
+
+(defmethod accept-spec 'clojure.core/float? [_ _ _ _] {:type "number" :format "float"})
+(defmethod accept-spec 'clojure.core/double? [_ _ _ _] {:type "number" :format "double"})
+(defmethod accept-spec 'clojure.core/nil? [_ _ _ _] {})
+
+(defmethod accept-spec 'clojure.spec.alpha/or [_ _ children _]
+  ;; :anyOf is not supported by Swagger 2.0, so we just the the first child. In principle,
+  ;; we could do better in some special cases. For example, a reasonable schema
+  ;; for (s/or ::int int? ::str string?) would be {:type ["number", "string"]}.
+  (assoc
+    (first children)
```

Comment:
The first value convention makes perfect sense for and-type conditions like `(s/conditional s/Int pos?)`, but not for or-type conditions.
* If your endpoint only takes a positive int and the docs say you can use any int, that's fine. If you send a negative int, you'll get hopefully-helpful error message.
* If your endpoint takes an int or a string, the docs say that you can only send an int and there's no way to discover that strings would be fine, too.

I didn't realize that type is required, but you're right. There's no good answer. Let's go with 1, then, at least it produces valid specs.

The Swagger 2.0 schema language is just too limited. Maybe OpenAPI 3.0 will save us.